fades/main.py

```diff
@@ -96,14 +96,33 @@ def consolidate_dependencies(needs_ipython, child_program,
 
 def decide_child_program(args_executable, args_child_program):
     """Decide which the child program really is (if any)."""
+    # We get the logger here because it's not defined at module level
+    logger = logging.getLogger('fades')
+
     if args_executable:
+        # if --exec given, check that it's just the executable name,
+        # not absolute or relative paths
+        if args_executable and os.path.sep in args_child_program:
+            logger.error(
+                "The parameter to --exec must be a file name (to be found "
+                "inside venv's bin directory), not a file path: %r",
+                args_child_program)
+            raise FadesError("File path given to --exec parameter")
+
         # indicated --execute, local and not analyzable for dependencies
         analyzable_child_program = None
         child_program = args_child_program
     elif args_child_program is not None:
         # normal case, the child program is to be analyzed (being it local or remote)
         if args_child_program.startswith(("http://", "https://")):
             args_child_program = helpers.download_remote_script(args_child_program)
+        else:
+            if not os.access(args_child_program, os.R_OK):
```

Comment:
Shouldn't you be checking for X_OK?

Reply:
mmmm we support a non executable file. I think read is ok.

Reply:
you're right